engine/api/v2_entities.go

```diff
@@ -77,3 +77,61 @@ func (api *API) getEntitiesHandler() ([]service.RbacChecker, service.Handler) {
 			return service.WriteJSON(w, result, http.StatusOK)
 		}
 }
+
+func (api *API) getEntityHandler() ([]service.RbacChecker, service.Handler) {
+	return service.RBAC(api.projectRead),
+		func(ctx context.Context, w http.ResponseWriter, req *http.Request) error {
+			vars := mux.Vars(req)
+			pKey := vars["projectKey"]
+			vcsIdentifier, err := url.PathUnescape(vars["vcsIdentifier"])
+			if err != nil {
+				return sdk.NewError(sdk.ErrWrongRequest, err)
+			}
+			repositoryIdentifier, err := url.PathUnescape(vars["repositoryIdentifier"])
+			if err != nil {
+				return sdk.WithStack(err)
+			}
+			entityType := vars["entityType"]
+			entityName := vars["entityName"]
+
+			branch := QueryString(req, "branch")
+
+			vcsProject, err := api.getVCSByIdentifier(ctx, pKey, vcsIdentifier)
+			if err != nil {
+				return err
+			}
+
+			repo, err := api.getRepositoryByIdentifier(ctx, vcsProject.ID, repositoryIdentifier)
+			if err != nil {
+				return err
+			}
+
+			if branch == "" {
+				tx, err := api.mustDB().Begin()
```

Comment:
why do you need a transaction to get the default branch ?

Reply:
VCS Client need a transaction since we implement bitbucketcloud